### PR TITLE
Upgrade outdated package dependencies

### LINKED ISSUE #353: Upgrade all the dev dependencies

### PR DESCRIPTION
**Pull request by danielotieno**
#### What does this PR do?
- Upgrades the outdated package dependencies. 

#### What issue does it fix?
- Fix #353

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,60 +35,60 @@
...
+    "rollup-plugin-uglify": "^6.0.3",
+    "typescript": "^3.6.3",
+    "webpack": "^4.41.0",
+    "webpack-cli": "^3.3.9",
+    "webpack-dev-server": "^3.8.2"
   },
   "peerDependencies": {
-    "react": "^0.14 || ^15.0.0-rc || ^15.0.0 || ^16.0.0-rc || ^16.0.0",
-    "react-dom": "^0.14 || ^15.0.0-rc || ^15.0.0 || ^16.0.0-rc || ^16.0.0"
+    "react": "^16.10",
```
Comment (s-yadav) [inline package.json]: We should maintain 15 version on peer dependency. Or else it might be a breaking change.

Comment (danielotieno) [reply package.json]: Thank you for the review

I will make the changes

**Comment by s-yadav** [comment]
Also, the Travis build is failing, so you might want to check that as well.

**Comment by s-yadav** [comment]
@danielotieno I guess updating node version to 8 in travis.yml should fix the issue.

**Comment by danielotieno** [comment]
Thank you for the tip 🎉

**Comment by danielotieno** [comment]
> @danielotieno I guess updating node version to 8 in travis.yml should fix the issue.

@s-yadav changing the node version does not help still failing 
```	Error: Expected '+1 (123) 678 9    US' to equal '+1 (123) 456 7 89 US'.
	    at <Jasmine>
	    at UserContext.<anonymous> (test/library/keypress_and_caret.spec.js:91013:37)
	    at <Jasmine>
	Error: Expected 14 to equal 17.
	    at <Jasmine>
	    at UserContext.<anonymous> (test/library/keypress_and_caret.spec.js:91014:24)
	    at <Jasmine>
```

**Comment by s-yadav** [comment]
Then it might be some API of those libraries might have changed. Interestingly the spec is failing which shouldn't have failed. Can you check what has changed on Jasmine with version upgrade.

**Comment by danielotieno** [comment]
> Then it might be some API of those libraries might have changed. Interestingly the spec is failing which shouldn't have failed. Can you check what has changed on Jasmine with the version upgrade?

Or I try to downgrade jasmine and jasmine-core to see if that's the one break the tests.

**Comment by danielotieno** [comment]
@s-yadav It's fixed

File: package.json [hunk 2]
```diff
@@ -1,96 +1,96 @@
...
-  },
-  "peerDependencies": {
-    "react": "^0.14 || ^15.0.0-rc || ^15.0.0 || ^16.0.0-rc || ^16.0.0",
-    "react-dom": "^0.14 || ^15.0.0-rc || ^15.0.0 || ^16.0.0-rc || ^16.0.0"
-  },
-  "dependencies": {
-    "prop-types": "^15.7.2"
-  }
-}
+    "name": "react-number-format",
```
Comment (s-yadav) [inline package.json]: Can you maintain the tab spacing to 2. ?

**Comment by danielotieno** [comment]
On it

File: package.json (hunk 2 above)
Comment (danielotieno) [reply package.json]: Done

**Comment by s-yadav** [comment]
Merged

**Pull request merged**